includes/class-wc-gateway-stripe.php

```diff
@@ -409,6 +409,19 @@ public function payment_scripts() {
 			'i18n_required_fields' => __( 'Please fill in required checkout fields first', 'woocommerce-gateway-stripe' ),
 		);
 
+		// If we're on the Add Payment Method page under My Account we'll create a SetupIntent and embed the client_secret in the JS.
+		if ( is_add_payment_method_page() ) {
+			$customer_object = new WC_Stripe_Customer( wp_get_current_user()->ID );
+
+			if ( empty( $customer_object ) ) {
+				// TODO: create a customer, because we absolutely need a customer object for the setup intent.
+			}
+
+			$setup_intent = WC_Stripe_API::request( [ 'customer' => $customer_object->get_id() ], 'setup_intents' );
+
+			$stripe_params['setup_intent_client_secret'] = $setup_intent->client_secret;
+		}
+
```

Comment:
This is part of where variables are loaded into the JS.

To limit the scope of this new section we only setup a setup intent when we're adding a new payment method from the my account page. This limits this functionality to just that page.

As I mentioned in a different comment we should probably abandon this approach in favor of setting up an API on the store that can proxy a request through to Stripe with the Source created in the frontend.

This at least shows how to get the Stripe Customer if it exists, and the general approach to creating the setup intent. Once we're doing this through an API we will need to return the result from that to the frontend so the frontend can decide on whether to prompt for SCA or not, similar to what we do when processing a regular payment.

Reply:
> As I mentioned in a different comment we should probably abandon this approach in favor of setting up an API on the store that can proxy a request through to Stripe with the Source created in the frontend.

And as mentioned [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1398#discussion_r543441034), I agree :)